mne/io/artemis123/artemis123.py

```diff
@@ -0,0 +1,217 @@
+
+    Returns
+    -------
+    raw : Instance of Raw
+        A Raw object containing the data.
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    return RawArtemis123(input_fname, preload=preload, verbose=verbose)
+
+
+def _get_artemis123_info(fname):
+    """Function for extracting info from artemis123 header files."""
+    fname = op.splitext(fname)[0]
+    header = fname + '.txt'
+
+    logger.info('Reading header...')
+
+    #key names for artemis channel info...
+    chanKeys = ['name','scaling','FLL_Gain','FLL_Mode','FLL_HighPass','FLL_AutoReset','FLL_ResetLock']
+
+    header_info = dict()
+    header_info['filterHist'] = [];
+    header_info['comments']   = ''
+    header_info['Channels']    = [];
+
+    with open(header, 'r') as fid:
+      #section flag
+      # 0 - None
+      # 1 - main header
+      # 2 - channel header
+      # 3 - comments
+      # 4 - length
+      # 5 - filtering History
+      sectionFlag = 0
+      for line in fid:
+        #skip emptylines or header line for channel info
+        if not line.strip() or ( sectionFlag==2 and line.startswith('DAQ Map') ):
+          continue
+
+        #set sectionFlag
+        if line.startswith('<end'):
+          sectionFlag = 0
+        elif line.startswith("<start main header>"):
+          sectionFlag = 1
+        elif line.startswith("<start per channel header>"):
+          sectionFlag = 2
+        elif line.startswith("<start comments>"):
+          sectionFlag = 3
+        elif line.startswith("<start length>"):
+          sectionFlag = 4
+        elif line.startswith("<start filtering history>"):
+          sectionFlag = 5
+        else:
+          #parse header info lines
+          if sectionFlag == 1: #part of main header - lines are name value pairs
+            values = line.strip().split('\t')
+            if len(values) == 1:
+              values.append('')
+            header_info[values[0]] = values[1]
+          elif sectionFlag == 2: #part of channel header - lines are Channel Info
+            values = line.strip().split('\t')
+            if len(values) != 7:
+              pass #error
+            tmp = dict()
+            for k,v in zip(chanKeys,values):
+              tmp[k] = v
+            header_info['Channels'].append(tmp)
+          elif sectionFlag == 3:
+            header_info['comments'] = '%s%s'%(header_info['comments'],line.strip())
+          elif sectionFlag == 4:
+            header_info['num_samples'] = int(line.strip())
+          elif sectionFlag == 5:
+            header_info['filterHist'].append(line.strip())
+
+    #########################################################################################
+    #build mne info struct
+    info = _empty_info(float(header_info['Rate Out']))
+
+    #Attempt to get time/date from fname
+    try:
+        date = datetime.datetime.strptime(op.basename(fname).split('_')[2],'%Y-%m-%d-%Hh-%Mm')
+    except:
```

Comment:
Do not `except:` but `except Exception:` because the former is too broad (will catch SystemExit)